streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -790,18 +790,30 @@ private void runLoop() {
     // Visible for testing
     long runOnce(final long recordsProcessedBeforeCommit) {
         long processedBeforeCommit = recordsProcessedBeforeCommit;
-        timerStartedMs = time.milliseconds();
 
-        // try to fetch some records if necessary
-        final ConsumerRecords<byte[], byte[]> records = pollRequests();
+        ConsumerRecords<byte[], byte[]> records;
+
+        timerStartedMs = time.milliseconds();
 
         if (state == State.PARTITIONS_ASSIGNED) {
+            // try to fetch some records with zero poll millis
```

Comment:
just to make sure I understand this correctly, this would only be fetching records for tasks that have completed restoration, correct?

Reply:
yes.